_topics/en/cluster-mempool.md

```diff
@@ -52,20 +55,102 @@ see_also:
 excerpt: >
   **Cluster mempool** is a proposal to associate each unconfirmed
   transaction in a mempool with related transactions, creating a
-  cluster.  Each cluster of transactions, whether it be a single
-  transaction or several transactions, can be ordered from most
-  desirable to mine to least desirable, allowing operations for adding
-  or removing new clusters to complete fast enough to use them in P2P
-  network code.
+  _cluster_.  Each cluster contains feerate-sorted _chunks_ consisting
+  of one or more transactions.  If clusters are limited in size,
+  we also limit how much needs to be recomputed in response to new
+  transactions being added to the mempool, allowing algorithmic
+  decisions affecting the entire mempool to complete fast enough to
+  use them in P2P network code.
 
 ---
-If P2P network code can determine whether a new unconfirmed transaction
-(or cluster of transactions) will improve the profitability of the
-mempool for miners in the near term, that criteria can be used instead
-of other heuristics for determining when a transaction [package][topic
-package relay] or [replacement][topic rbf] should be accepted.  It is
-hoped that this will allow transaction relay to provide more flexible
-policy without exposing relay nodes to new resource-wasting attacks.
+The overall goal of cluster mempool is being able to reason about the
+effect of transactions on the blocks a miner would create if it has an
+identical mempool to the local node's mempool.
+
+The most apparent example for why we have a need for that kind of
+reasoning is the fact that today, the eviction mechanism (when the
+mempool exceeds the node's size limit) can choose to evict the very best
+transaction in the mempool overall ([example][wuille slide7]).
+
+To describe the current limitation in short: miners prefer to select the
```

Comment:
```suggestion
To describe the current limitation in short: miners prefer to select for inclusion the
```
(otherwise, at first I read this as selecting for eviction, since that was just being mentioned.)